sql/src/main/java/io/crate/operation/collect/files/FileReadingCollector.java

```diff
@@ -48,11 +51,14 @@
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.*;
+import java.util.concurrent.CompletableFuture;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 import java.util.zip.GZIPInputStream;
 
-public class FileReadingCollector implements CrateCollector {
+import static io.crate.exceptions.Exceptions.rethrowUnchecked;
+
+public class FileReadingCollector implements BatchIterator, CrateCollector {
```

Comment:
i think an inner class for implementing the BatchIterator would make sense for readability